firewood/src/merkle.rs

```diff
@@ -465,6 +420,87 @@ impl<S: ShaleStore<Node> + Send + Sync> Merkle<S> {
                     break None;
                 }
 
+                NodeType::Branch(n) if n.path.len() == 0 => {
```

Comment:
Can you move this back? I can't easily tell what changed. Alternatively, do the move in one commit and then make the changes in another.